File: bridge/samples/hello_world/capabilities/greeting/compose/executor.py
"""Executor factory for greeting.compose (contracts/greeting.compose.contract.yaml).

executor_kind: factory -- make_executor is called once, at assembly
time, with a Dependencies scoped to dependencies.capabilities
(console.write, R4). Resolves console.write once and returns the
execute(operation, input, policy) closure; every call is a real,
fully-traced Bridge request (R6/R8), never a direct executor call.

`name` isn't checked here -- Bridge.handle already validated it
(required, type: string, non-blank via `pattern: "\S"`, not minLength).
"""


def make_executor(dependencies):
    write = dependencies.resolve("console.write", "write")

    def execute(operation, input, policy):
        write.call({"text": f"Greetings, {input['name']}!"}, policy_context=policy)
        return {}

    return execute

File: sample/hello_world/frontend/runtime/capabilities/greeting/render/executor.js
"use strict";
/**
 * Executor factory for greeting.render (../../../contracts/greeting.render.contract.yaml),
 * Local execution (executor_kind: factory) in the frontend's own Bridge
 * Core -- its OWN capability, not a second implementation of backend's
 * greeting.compose (see the contract's own header for why). makeExecutor
 * is called once, at assembly time, with a Dependencies scoped to
 * dependencies.capabilities (console.write, R4). console.write only
 * exists Remote from this runtime's own registry (../../console/write/manifest.json) --
 * this factory doesn't know or care; `write.call(...)` is a real,
 * fully-traced Bridge request either way (R6/R8).
 */

function makeExecutor(dependencies) {
  const write = dependencies.resolve("console.write", "write");

  return async function execute(operation, input, policy) {
    const message = `Greetings, ${input.name}! (rendered by the frontend)`;
    // console.write is pinned <2.0.0 (this contract's own
    // dependencies.capabilities) -- 1.0.0's `text` field, not 2.0.0's `message`.
    await write.call({ text: message }, policy);
    return { message };
  };
}

export { makeExecutor };
